projects/plugins/crm/api/create_company.php

```diff
@@ -97,156 +85,144 @@
+		$update_args['id'] = $verified_id;
+	}
 
-						apply_filters('post_do_this', $args);  // etc..
+	// } Tags
+	if ( $we_have_tags ) {
+		$update_args['tags'] = $company_tags;
+	}
 
-					*/
+	// } Status default - double-backup for api check
+	if ( isset( $update_args ) && is_array( $update_args ) && ( ( $update_args[ $field_prefix . 'status' ] ) === null || ! isset( $update_args[ $field_prefix . 'status' ] ) || empty( $update_args[ $field_prefix . 'status' ] ) ) ) {
+		$default_status                          = zeroBSCRM_getSetting( 'defaultstatus' );
+		$update_args[ $field_prefix . 'status' ] = $default_status; // 'Lead';
+	}
 
-					$newCompany = zeroBS_integrations_addOrUpdateCompany(
-						'api',
-						$email,
-						$update_args,
-						'', // ) Customer date (auto)
-						$fallBackLog, // } Fallback log (for customers who already exist)
-						false, // } Extra meta
-						$internalAutomatorOverride, // } Internal automator overrides - here we pass a "customer.create" note override (so we can pass it a custom str, else we let it fall back to "created by API")
-						'update',
-						$fieldPrefix // field prefix zbsc_
-					);
-					// ^^ this'll be either: ID if added, no of rows if updated, or FALSE if failed to insert/update
+	/**
+	 * Need to pass via the update_args otherwise the tags are added AFTER the automation fires...      when doing new DB we need to hook and filter up varios steps of these
+	 * e.g.
+	 * apply_filters('pre_do_this', $args);
+	 * ...do_this...
+	 * apply_filters('post_do_this', $args);  // etc..
+	 */
+
+	$new_company = zeroBS_integrations_addOrUpdateCompany(
+		'api',
+		$email,
+		$update_args,
+		'', // ) Customer date (auto)
+		$fallback_log, // } Fallback log (for customers who already exist)
+		false, // } Extra meta
+		$internal_automator_override, // } Internal automator overrides - here we pass a "customer.create" note override (so we can pass it a custom str, else we let it fall back to "created by API")
+		'update',
+		$field_prefix // field prefix zbsc_
+	);
+	// ^^ this'll be either: ID if added, no of rows if updated, or FALSE if failed to insert/update
 
-					// } are we assigning to a user?
-					if ( isset( $assign ) && ! empty( $assign ) ) {
-						// set owner
-						zeroBS_setOwner( $newCompany, $assign, ZBS_TYPE_COMPANY );
-					}
+	// } are we assigning to a user?
+	if ( ! empty( $assign ) ) {
+		// set owner
+		zeroBS_setOwner( $new_company, $assign, ZBS_TYPE_COMPANY );
+	}
 
-					// old way just returned what was sent...
-					// wp_send_json($json_params); //sends back to Zapier the customer that's been sent to it.
+	// old way just returned what was sent...
+	// wp_send_json($json_params); //sends back to Zapier the customer that's been sent to it.
 
-					// thorough much? lol.
-					if ( ! empty( $newCompany ) && $newCompany !== false && $newCompany !== -1 ) {
+	// thorough much? lol.
+	if ( ! empty( $new_company ) && $new_company !== false && $new_company !== -1 ) {
```

Comment:
I'm surprised Phan didn't flag this. If `! empty( $new_company )`, then it's certainly also `!== false`.
```suggestion
	if ( ! empty( $new_company ) && $new_company !== -1 ) {
```

Reply:
bf9dad7aa2c